graylog2-web-interface/src/views/pages/NewSearchPage.test.jsx

```diff
@@ -0,0 +1,194 @@
+    expect(getByText('Loading...')).not.toBeNull();
+  });
+
+  describe('mounting', () => {
+    it('should create new view with type search', async () => {
+      const createViewAction = asMock(ViewActions.create);
+
+      render(<NewSearchPage route={{}}
+                            router={mockRouter}
+                            location={{}} />);
+
+      await wait(() => expect(createViewAction).toBeCalledTimes(1));
+      await wait(() => expect(createViewAction).toHaveBeenCalledWith(View.Type.Search));
+    });
+
+    it('should process hooks with provided location query', async () => {
+      const processHooksAction = asMock(processHooks);
+
+      render(<NewSearchPage route={{}}
+                            router={mockRouter}
+                            location={mockLocation} />);
+
+      await wait(() => expect(processHooksAction).toBeCalledTimes(1));
+      await wait(() => expect(processHooksAction.mock.calls[0][3]).toStrictEqual({ q: '', rangetype: 'relative', relative: '300' }));
+    });
+  });
+
+  describe('loading another view', () => {
+    it('should be possible with specific view id', async () => {
+      mockExtendedSearchPage.mockImplementation(() => (
+        <ViewLoaderContext.Consumer>
+          {loadView => <button type="button" onClick={() => loadView && loadView('special-view-id')}>Load view</button>}
+        </ViewLoaderContext.Consumer>
+      ));
+      const viewGetAction = asMock(ViewManagementActions.get);
+
+      const { getByText } = render(<NewSearchPage route={{}}
+                                                  router={mockRouter}
+                                                  location={{}} />);
+      const viewLoadButton = await waitForElement(() => getByText('Load view'));
+      fireEvent.click(viewLoadButton);
+
+      await wait(() => expect(viewGetAction).toHaveBeenCalledTimes(1));
+      await wait(() => expect(viewGetAction).toHaveBeenCalledWith('special-view-id'));
+    });
+
+    // it('should show error message when loading fails', async () => {
+    //   const hookComponent = <div>The hook component</div>;
+    //   ViewManagementActions.get = mockAction(jest.fn(() => Promise.reject(hookComponent)));
+    //   mockExtendedSearchPage.mockImplementation(() => (
+    //     <ViewLoaderContext.Consumer>
+    //       {loadView => loadView && <button type="button" onClick={() => loadView('special-view-id')}>Load view</button>}
+    //     </ViewLoaderContext.Consumer>
+    //   ));
+
+    //   const { getByText } = render(<NewSearchPage route={{}}
+    //                                               router={mockRouter}
+    //                                               location={{}} />);
+    //   const viewLoadButton = await waitForElement(() => getByText('Load view'));
+    //   fireEvent.click(viewLoadButton);
+
+    //   await waitForElement(() => getByText('The hook component'));
+    // });
+  });
```

Comment:
Question: Having a look at the implementation of `processHooks`, I realized we should unify the usage, especially in the context of the error case. Sometimes we are catching the error and creating a `UserNotification`:
```
 return processHooks(
      ...
 ).catch(
     error => UserNotification.error(`Executing search failed with error: ${error}`, 'Could not execute search'),
);
```
sometimes we are using the `processHooks` `onError` argument, to set the `hookComponent` state 
```
return viewLoader(
      ...
      (e) => {
        if (e instanceof Error) {
          throw e;
        }
        this.setState({ hookComponent: e });
      },
```
and sometimes we are doing the same, but without checking the error instance.

My main question is related to the `hookComponent`. It looks like we are defining a `hookComponent` when `processHooks` errors. When the `hookComponent` is defined we'll display it instead of the `ExtendedSearchPage`. I want to test tis case, but looking at the code I don't understand when the `hookComponent` (e.g. in the last code snippet) gets defined. because when it is an error, `setState` will not get called. When it is not an error, the `onError` function passed to `processHooks` will not getCalled.